packages/react-scripts/scripts/build.js

```diff
@@ -141,3 +142,30 @@ function copyPublicFolder() {
     filter: file => file !== paths.appHtml,
   });
 }
+
+function formatError(err) {
+  const message = get(err, 'message');
+
+  // Add more helpful message for UglifyJs error
+  if (typeof message === 'string' && message.indexOf('from UglifyJs') !== -1) {
+    try {
+      console.log(
+        'UglifyJs could not parse the code from \n\n',
+        chalk.yellow(
+          err.stack.split('\n')[1].split('[')[1].split('][')[0].replace(']', '')
```

Comment:
I would prefer if we checked against a regex (and extracted from it) instead.
So that if it doesn't match, we don't attempt to parse.